File: 2023/challenge_02/challenge_02.py
def run():
  input_value = '&###@&*&###@@##@##&######@@#####@#@#@#@##@@@@@@@@@@@@@@@*&&@@@@@@@@@####@@@@@@@@@#########&#&##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@&'
  value = 0
  result = '' 

  for caracter in input_value:
    match caracter:
      case '#':
        value += 1
      case '@':
        value -= 1
      case '*':
        value *= value
      case '&':
          result += str(value)
  
  return result

if __name__ == '__main__':
  print(run())

File: 2023/challenge_02/challenge_02.js
function run() {
  const MESSAGE = '&###@&*&###@@##@##&######@@#####@#@#@#@##@@@@@@@@@@@@@@@*&&@@@@@@@@@####@@@@@@@@@#########&#&##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@&';
  let value = 0;
  let result = '';
  const CASES = {
    '#': () => value += 1,
    '@': () => value -= 1,
    '*': () => value *= value,
    '&': () => result = result.concat(value)
  }

  MESSAGE.split('').forEach(element => CASES[element]?.());
  return result;
}

console.log(run())